### PR TITLE
Implement client for experimental-features API

### PR DESCRIPTION
**Pull request by hmacr**
# Pull Request

## Related issue
Needed for #498 & #501, since these issues involve updating the experimental-feature as a pre-requisite

## What does this PR do?
- Introduces the experimental-features client in the SDK
- Implements the `get` and `patch` endpoint.
- Adds code-samples and tests.

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by Kerollmops — DISMISSED** [review]
Looks good to me ❤️ 
I'll let @irevoire merge!

**Review by irevoire — APPROVED** [review]
bors merge

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -1656,3 +1656,19 @@ multi_search_1: |-
...
+    .unwrap();
+update_experimental_features_1: |-
+  let client = Client::new("http://localhost:7700", Some("apiKey"));
+  let mut features = ExperimentalFeatures::new(&client);
+  features.set_score_details(true);
+  features.set_vector_store(true);
+  let res = features
+    .update()
+    .await
+    .unwrap();
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:1674]:
```suggestion
    .unwrap();

```

**Review by curquiza — APPROVED** [review]
bors merge

**Comment by curquiza** [comment]
bors merge

**Pull request merged**